modules/apps/translation/translation-service/src/main/java/com/liferay/translation/internal/importer/XLIFFInfoFormTranslationImporter.java

```diff
@@ -83,604 +32,27 @@
-				targetLanguage);
-		}
-	}
-
-	private void _validateXLIFFDocumentName(
-			InfoItemReference infoItemReference, String xliffDocumentName)
-		throws XLIFFFileException {
-
-		if (infoItemReference == null) {
-			return;
-		}
-
-		Matcher matcher = _pattern.matcher(xliffDocumentName);
-
-		if (!matcher.matches()) {
-			throw new XLIFFFileException.MustHaveValidId("File ID is invalid");
-		}
-
-		String className = matcher.group(1);
-		long classPK = GetterUtil.getLong(matcher.group(2));
-
-		if (Objects.equals(className, infoItemReference.getClassName()) &&
-			Objects.equals(classPK, infoItemReference.getClassPK())) {
-
-			return;
-		}
-
-		if (!Objects.equals(className, SegmentsExperience.class.getName())) {
-			throw new XLIFFFileException.MustHaveValidId("File ID is invalid");
-		}
-
-		SegmentsExperience segmentsExperience =
-			_segmentsExperienceLocalService.fetchSegmentsExperience(classPK);
-
-		if ((segmentsExperience == null) ||
-			!Objects.equals(
-				segmentsExperience.getPlid(),
-				_getSegmentsExperienceClassPK(infoItemReference))) {
-
-			throw new XLIFFFileException.MustHaveValidId("File ID is invalid");
-		}
-	}
-
-	private void _validateXLIFFFile(
-			long groupId, InfoItemReference infoItemReference,
-			XLIFFDocument xliffDocument)
-		throws XLIFFFileException {
-
-		_validateXLIFFCompletion(groupId, xliffDocument);
-
-		_validateXLIFFFileNode(infoItemReference, xliffDocument);
-	}
-
-	private void _validateXLIFFFileNode(
-			InfoItemReference infoItemReference, XLIFFDocument xliffDocument)
-		throws XLIFFFileException {
-
-		List<String> fileNodeIds = xliffDocument.getFileNodeIds();
-
-		if (fileNodeIds.size() != 1) {
-			throw new XLIFFFileException.MustNotHaveMoreThanOne(
-				"Only one node is allowed");
-		}
-
-		_validateXLIFFDocumentName(infoItemReference, fileNodeIds.get(0));
-	}
-
-	private void _validateXLIFFStartSubdocument(
-			InfoItemReference infoItemReference,
-			StartSubDocument startSubDocument)
-		throws XLIFFFileException {
-
-		if (startSubDocument == null) {
-			throw new XLIFFFileException.MustBeWellFormed(
-				"The XLIFF file is not well Formed");
-		}
-
-		_validateXLIFFDocumentName(
-			infoItemReference, startSubDocument.getName());
-
-		Property targetLanguageProperty = startSubDocument.getProperty(
-			"targetLanguage");
-
-		if ((targetLanguageProperty == null) ||
-			(targetLanguageProperty.getValue() == null)) {
-
-			throw new XLIFFFileException.MustBeWellFormed(
-				"There is no translation target");
-		}
-	}
-
-	private static final Pattern _pattern = Pattern.compile("([^:]+):(.+)");
-
-	@Reference
-	private Language _language;
-
-	@Reference
-	private LayoutLocalService _layoutLocalService;
-
-	@Reference
-	private SAXReader _saxReader;
-
-	@Reference
-	private SegmentsExperienceLocalService _segmentsExperienceLocalService;
+	@Reference(
+		target = "(component.name=com.liferay.translation.internal.importer.XLIFFTranslationSnapshotProvider)"
```

Comment:
@Valmir-unicap Based on other reference (ImportTranslationMVCActionCommand) to `TranslationSnapshotProvider`, we can assume that it has only 1 service. So this reference can be without the `target` part.